### PR TITLE
Clean up code/solution

### PR DESCRIPTION
**Pull request by oliverjam**
1. Switch challenge to use arrow functions for anonymous callbacks (they're a lot easier to read)
1. Put full solution in standalone js file (with comments explaining each step)
1. Replace solution README (which was just a copy/paste of the main README) with an actual step-by-step explanation of how the recursive solution works

### CODE REVIEW

File: solution/README.md [hunk 1]
```diff
@@ -1,88 +1,115 @@
-# Morning challenge: implement a waterfall function
+# Solution explanation
```
Comment (astroash) [inline solution/README.md:1]: This is the only workshop I know of to have a full written explanation in the solutions folder. It's worth mentioning this in the main readme so students know it's here.

**Review by astroash — CHANGES_REQUESTED** [review]

File: solution/README.md (hunk 1 above)
Comment (oliverjam) [reply solution/README.md:1]: Yeah true. I was planning on rewriting the readme at some point as well, but I'll drop a link in for now

Comment (oliverjam) [reply solution/README.md:1]: Done

**Review by astroash — APPROVED** [review]
✨

**Review by shiryz — APPROVED** [review]
Looks great, thanks @oliverjam :shipit:

**Pull request merged**